src/languages/php.js

```diff
@@ -52,12 +52,12 @@ export default function(hljs) {
     contains: hljs.QUOTE_STRING_MODE.contains.concat(SUBST),
   });
   const HEREDOC = hljs.END_SAME_AS_BEGIN({
-    begin: /<<<[ \t]*(\w+)\n/,
+    begin: /<<<[ \t]*"?(\w+)"?\n/,
```

Comment:
This allows:

`"BOO`

`BOO"`

You need to use an OR (like I showed you before) to make sure they are balanced.

Reply:
`"` itself must not be included in match. For now I don't know how to do that. `(\w+|"\w+")` this is not work.

I don't think it's a problem. It's just highlighter but not PHP code parser.

Reply:
`<<<[ \t]*(?:(\w+)|"(\w+)")` don't works as well because hljs.END_SAME_AS_BEGIN works with first group only...

Reply:
Any ideas how to solve this?

Reply:
Make the outside group capturing and the two inside ones non-capturing - though technically you don't need the inside groups at all the `|` and outside group should be fine by themselves.

Reply:
But then an outside group takes over the `"` sign and we don't need it - it breaks everything.

![image](https://user-images.githubusercontent.com/15121552/208278285-eab16185-061b-4a43-b54d-3f2bfa985358.png)

https://regex101.com/r/I0FG3H/1

Reply:
And even more we have one more problem here 
![image](https://user-images.githubusercontent.com/15121552/208278327-7a87c9ea-6e98-4180-a876-0264315a2e71.png)

Reply:
The only solution I found - is rewrite `hljs.END_SAME_AS_BEGIN` and then `(?:(\w+)|"(\w+)")` works good:

```js
  const HEREDOC = {
    begin: /<<<[ \t]*(?:(\w+)|"(\w+)")\n/,
    end: /[ \t]*(\w+)\b/,
    contains: hljs.QUOTE_STRING_MODE.contains.concat(SUBST),
    'on:begin': (m, resp) => { resp.data._beginMatch = m[1] || m[2]; },
    'on:end': (m, resp) => { if (resp.data._beginMatch !== m[1]) resp.ignoreMatch(); },
  };
```

Reply:
Yeah, I was just about to mention it's just JS under the covers and you could tweak it like that... :-) Looks good enough to me.

Reply:
But I think it's better to leave `begin: /<<<[ \t]*"?(\w+)"?\n/,`. 

That's okay about `"BOO` & `BOO"`. Because we, in other hand, don't check the full workability of the HEREdoc construct, which implies that the indentation of the text will be exactly the same as the last END.

Reply:
When it's easy to avoid false positives with a strong regex, we go that route... which is this case means matching quoted OR unquoted, not half quoted.